lib/datadog/statsd.rb

```diff
@@ -33,14 +83,23 @@ class Connection
       # DogStatsd unix socket path. Not used by default.
       attr_reader :socket_path
 
+      def initialize(telemetry)
+        @telemetry = telemetry
+      end
+
       # Close the underlying socket
       def close
         @socket && @socket.close
       end
 
       def write(message)
         @logger.debug { "Statsd: #{message}" } if @logger
-        send_message(message)
+        payload = message + @telemetry.flush()
+        send_message(payload)
+
+        @telemetry.reset
+        @telemetry.bytes_sent += payload.length
```

Comment:
Here we reset the values and then we're setting `bytes_sent` and incrementing `packets_sent`. Doesn't it mean that these values will always be equal to what we set them to here? (they're not modified anywhere apart from here and the reset method).

Reply:
yes